lib/grant.js

```diff
@@ -151,7 +151,7 @@ function checkClient (done) {
  * @this   OAuth
  */
 function checkGrantType (done) {
-  if (this.grantType.match(/^http(s|):\/\//) && this.model.extendedGrant) {
```

Comment:
Checking for the scheme part is a good idea IMHO. I just had a look at the URI spec at https://tools.ietf.org/html/rfc3986#section-3.1 , there the scheme is defined as:

```
scheme      = ALPHA *( ALPHA / DIGIT / "+" / "-" / "." )
```
So the regex should probably be something like /^[a-zA-Z][a-zA-Z0-9+.-]+:/